### PR TITLE
Nightly build with GitHub actions

### PR DESCRIPTION
**Pull request by ja2-derek**
Drafting this to share the work for #1276 that I have done so far. It does not work yet but I have no way to test it right now.

The coverity build flow is taken from  https://scan.coverity.com/scripts/travisci_build_coverity_scan.sh; I can't find any official instructions for GitHub actions.

----

This PR assumes the presence of 2 repository/org secrets:
- `BOT_GITHUB_TOKEN` which is a user access token belonging to @madlab-bot with permissions to merge commits into `nightly`
- `COVERITY_SCAN_TOKEN` which is the token to submit Coverity builds

Can @selaux please to add those as [organization secrets](https://docs.github.com/en/free-pro-team@latest/actions/reference/encrypted-secrets#creating-encrypted-secrets-for-an-organization)?

### CODE REVIEW

File: .github/workflows/coverity-scan.yml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+                -O cov-analysis-linux64.tar.gz
+          mkdir cov-analysis-linux64
+          tar xzf cov-analysis-linux64.tar.gz --strip 1 -C cov-analysis-linux64
+          export PATH=$(pwd)/cov-analysis-linux64/bin:$PATH
+
+      - name: Fixed world writable dirs
+        run: |
+          chmod go-w $HOME
+          sudo chmod -R go-w /usr/share
+
```
Comment (lynxlynxlynx) [inline .github/workflows/coverity-scan.yml:37]: Why is this needed?

**Comment by lynxlynxlynx** [comment]
The coverity stuff looks fine. I wrote this last year and it pretty much matches:
https://github.com/gemrb/gemrb/blob/master/.github/workflows/coverity.yml

File: .github/workflows/coverity-scan.yml (hunk 1 above)
Comment (ja2-derek) [reply .github/workflows/coverity-scan.yml:37]: I just copied it somewhere without thinking too much. I will remove that.

**Comment by selaux** [comment]
I added the coverity token. For `BOT_GITHUB_TOKEN` this might take some time as I have to remember the credentials first.

**Comment by selaux** [comment]
Allright. The required secrets should be there now.

**Comment by ja2-derek** [comment]
> Allright. The required secrets should be there now.

@selaux can you check if there is an extra space in the `BOT_GITHUB_TOKEN`?

**Comment by selaux** [comment]
@ja2-derek There was a leading space, should be fixed now.

**Comment by ja2-derek** [comment]
The new build workflows are working - [nightly trigger](https://github.com/ja2-stracciatella/ja2-stracciatella/actions/runs/376026603) and [Coverity scan](https://github.com/ja2-stracciatella/ja2-stracciatella/runs/1435856212?).

A couple more changes:
- Added a CMake variable to skip `sccache`, which is required for Coverity build to work
- No longer using the `ad-m/github-push-action` action. We are doing a straight-forward `git-push` but with custom credentials. It is easier to do it with just shell commands.

----

But I don't understand why the Mac CI build is now failing the locate `gcloud`. I will take a look tomorrow.

**Comment by lynxlynxlynx** [comment]
LGTM. I know how annoying it can be to get these working, so kudos!

**Comment by ja2-derek** [comment]
The Mac build failure is due to a [recent change](https://github.blog/changelog/2020-10-01-github-actions-deprecating-set-env-and-add-path-commands/) from GitHub. It should be fixed now.

This PR is ready for merge. I will create another PR to remove `.travis.yml` after confirming builds works in master.
